### PR TITLE
Rewrites the "Labels" documentation for clarity and flow.

### PR DESCRIPTION
**Pull request by sergetron**
# Changes

Rewrites the "Labels" documentation for clarity and flow.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes
```release-note
   NONE
```

### CODE REVIEW

File: docs/labels.md [hunk 1]
```diff
@@ -6,88 +6,106 @@ weight: 10
...
-## Examples
-
-- [Finding Pods for a Specific PipelineRun](#finding-pods-for-a-specific-pipelinerun)
-- [Finding TaskRuns for a Specific Task](#finding-taskruns-for-a-specific-task)
-
-### Finding Pods for a Specific PipelineRun
-
-To find all `Pods` created by a `PipelineRun` named test-pipelinerun, you could
-use the following command:
+## Label propagation
```
Comment (dlorenc) [inline docs/labels.md:22]: One general question that might be good to answer here:

What happens if a label already exists? Do we stomp over it?

I think we preserve the existing one, but would have to confirm.

File: docs/labels.md [hunk 2]
```diff
@@ -6,88 +6,106 @@ weight: 10
...
+
+- For standalone `TaskRuns` (that is, ones not executing as part of a `Pipeline`), labels
+propagate from the [referenced `Task`](taskruns.md#specifying-the-target-task), if one exists, to
+the corresponding `TaskRun`, and then to the associated `Pod`.
+
+- For `Conditions`, labels propagate to the corresponding `TaskRuns`, and then to the associated `Pods`.
+
+## Automatic labeling
+
+Tekton automatically addes the following labels to Tekton entities:
```
Comment (dlorenc) [inline docs/labels.md:41]: Maybe it's worth adding a general statement that labels prefixed with "*.tekton.dev should be considered reserved for use by tekton, and should not be used by users.

File: docs/labels.md [hunk 3]
```diff
@@ -6,88 +6,106 @@ weight: 10
...
+## Label propagation
+
+Labels propagate among Tekton entities as follows:
+
+- For `Pipelines` instantiated using a `PipelineRun`, labels propagate
+automatically from `Pipelines` to `PipelineRuns` to `TaskRuns`, and then to
+the associated `Pods`.
+
+- Labels from `Tasks` referenced by `TaskRuns` within a `PipelineRun` propagate to the corresponding `TaskRuns`,
+and then to the associted `Pods`.
```
[the remark this answers is not in the document]
Comment (sergetron) [reply docs/labels.md:31]: Fixed.

File: docs/labels.md (hunk 1 above)
Comment (sergetron) [reply docs/labels.md:22]: I'll be happy to add a clarification. Can someone verify this?

File: docs/labels.md (hunk 2 above)
Comment (sergetron) [reply docs/labels.md:41]: Added a note that clarifies this.

File: docs/labels.md (hunk 1 above)
Comment (pritidesai) [reply docs/labels.md:22]: > What happens if a label already exists? Do we stomp over it?

we overwrite existing labels 😢 see open question in issue #2934

> E.g. For standalone `TaskRuns` (that is, ones not executing as part of a `Pipeline`), labels
propagate from the [referenced `Task`](taskruns.md#specifying-the-target-task), if one exists, to
the corresponding `TaskRun`, and then to the associated `Pod`. 

Labels from referenced `Task` propagate to `TaskRun` and overwrites if same labels exist in `TaskRun`.

I seems bigger discussion and worth bringing it to API Working group 😸

File: docs/labels.md [hunk 4]
```diff
@@ -6,88 +6,108 @@ weight: 10
...
+propagate from the [referenced `Task`](taskruns.md#specifying-the-target-task), if one exists, to
+the corresponding `TaskRun`, and then to the associated `Pod`.
+
+- For `Conditions`, labels propagate to the corresponding `TaskRuns`, and then to the associated `Pods`.
+
+## Automatic labeling
+
+Tekton automatically adds labels to Tekton entities as described in the following table.
+
+**Note:** `*.tekton.dev` labels are reserved for Tekton's internal use only. Do not add or remove them manually.
```
Comment (pritidesai) [inline docs/labels.md:43]: PR #2891 is enforcing this 🛑

File: docs/labels.md (hunk 1 above)
Comment (sergetron) [reply docs/labels.md:22]: Thanks! Let's make another PR after the discussion takes place and merge this PR in the meantime.

File: docs/labels.md (hunk 4 above)
Comment (sergetron) [reply docs/labels.md:43]: Do we need to modify this note?

File: docs/labels.md (hunk 1 above)
Comment (pritidesai) [reply docs/labels.md:22]: We will be discussing this in API WG on 7/20 😄
yup @sergetron we can create separate PR after the discussion.

File: docs/labels.md (hunk 4 above)
Comment (pritidesai) [reply docs/labels.md:43]: Nope

File: docs/labels.md (hunk 1 above)
Comment (sergetron) [reply docs/labels.md:22]: Excellent, resolving for now!

File: docs/labels.md (hunk 4 above)
Comment (sergetron) [reply docs/labels.md:43]: Great. :) Resolving.

**Comment by sergetron** [comment]
Can we have an LGTM please? :D

File: docs/labels.md [hunk 5]
```diff
@@ -6,88 +6,108 @@ weight: 10
 -->
 # Labels
 
-In order to make it easier to identify objects that are all part of the same
-conceptual pipeline, custom
-[labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)
-set on resources used by Tekton Pipelines are propagated from more general to
-more specific resources, and a few labels are automatically added to make it
-easier to identify relationships between those resources.
+Tekton allows you to use custom [Kubernetes Labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)
```
Comment (pritidesai) [inline docs/labels.md:9]: NIT: I don't have strong opinion here but might be better to keep it [labels] instead of making it [Kubernetes Labels] 🤔

Comment (sergetron) [reply docs/labels.md:9]: I qualified it as Kubernetes labels to make it clear we're using a pre-existing entity/concept here and didn't invent our own "label" thing from scratch. If that's confusing, I can revert it.

**Comment by pritidesai** [comment]
/lgtm

Based on today's API WG discussion, we are creating a separate doc in `community` repo to list all the possible labels getting created with "*.tekton.dev" which can be reference here (in a separate PR) and leaving the enforcement limited to documentation.

Also, the label ordering needs further evaluation before making changes (issue #2934 and #2121)

**Pull request merged**